server/createComponentRouter.js

```diff
@@ -0,0 +1,29 @@
+var React = require('react');
+var renderToString = require('react-dom/server').renderToString;
+var ReactRouter = require('react-router');
+
+var match = ReactRouter.match;
+var RoutingContext = ReactRouter.RoutingContext;
+
+module.exports = function createComponentRouter (routes) {
+  return function componentRouter (req, res, next) {
+    match({ routes: routes, location: req.url },
+        function (error, redirectLocation, renderProps) {
+
+      if (error) return res.status(500).send(error.message);
```

Comment:
> what do you mean?

I assume Keystone supports rendering custom error messages -- the line above will send a pretty generic error.

> Also are there mixed tabs & spaces in this file? the indentation in the snippet above looks weird

Maybe, but lint passes. I'll check it out. Is there a lint rule to cover that? And did `.editorconfig` fall asleep at the wheel?